src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/SubaccountTransactionSupervisor.kt

```diff
@@ -147,6 +148,36 @@ internal class SubaccountTransactionSupervisor(
         return submitCancelOrder(orderId, marketId, callback, payload, analyticsPayload, uiClickTimeMs)
     }
 
+    internal fun cancelAllOrders(marketId: String?, callback: TransactionCallback): HumanReadableCancelAllOrdersPayload {
+        val payload = payloadProvider.cancelAllOrdersPayload(marketId)
+
+        payload.payloads.forEach { cancelPayload ->
+            val subaccount = stateMachine.state?.subaccount(subaccountNumber)
```

Comment:
nit: `val subaccount = stateMachine.state?.subaccount(subaccountNumber) ?: return null` maybe? since subaccount should never be null

Reply:
that's why the exception is thrown!

Reply:
but actually u right i can move this up!!!